### ISSUE TITLE
Fix accessibility issues in the documentation

### ISSUE THREAD

**Issue by oss6** [r-devel/r-dev-env#308]
## Summary

The documentation should be accessible to everyone. Therefore, we should address any accessibility issues that can be fixed directly in the codebase. For issues that originate upstream, open the relevant issues with the appropriate maintainers.

## Technical details

Using a script that runs `axe` accessibility checks against every page of the MkDocs site (I can share the script so that it's a routine check) we get the following:

```
Scanned 20 pages: 2 rules violated, 78 nodes affected

  serious   color-contrast            52 nodes   20 pages
            Elements must meet minimum color contrast ratio thresholds
              /
              /container_setup/
              /container_setup/github_codespace/codespacestartstop/
              /container_setup/github_codespace/creating_codespace/
              /container_setup/github_codespace/live_share/
              /container_setup/gitpod_workspace/live_share/
              /container_setup/gitpod_workspace/workspacestart/
              /container_setup/gitpod_workspace/workspacestop_and_restart/
              /container_setup/local_setup/localsetup/
              /contributor_guide/contributing_to_codebase/
              /contributor_guide/contributing_to_docs/
              /resources/
              /troubleshoot/
              /tutorials/building_r/
              /tutorials/contribution_workflow/
              /tutorials/multi_r_compilation/
              /tutorials/patch_update/
              /tutorials/running_r/
              /tutorials/svn_help/
              /tutorials/update_source/
            https://dequeuniversity.com/rules/axe/4.13/color-contrast?application=webdriverjs

  serious   link-in-text-block        26 nodes   12 pages
            Links must be distinguishable without relying on color
              /container_setup/
              /container_setup/github_codespace/codespacestartstop/
              /container_setup/github_codespace/creating_codespace/
              /container_setup/gitpod_workspace/workspacestart/
              /container_setup/local_setup/localsetup/
              /contributor_guide/contributing_to_docs/
              /tutorials/building_r/
              /tutorials/contribution_workflow/
              /tutorials/multi_r_compilation/
              /tutorials/patch_update/
              /tutorials/svn_help/
              /tutorials/update_source/
            https://dequeuniversity.com/rules/axe/4.13/link-in-text-block?application=webdriverjs
```

See the full report here:

[a11y-report.json](https://github.com/user-attachments/files/31693697/a11y-report.json)

**Comment by soph-scott** [comment]
Hi @oss6, I can take a look at these. Would you mind assigning me to the issue if you can?

**Comment by oss6** [comment]
Hi @soph-scott yes please go ahead! Sorry I can't seem to change the assignee list.

**Comment by soph-scott** [comment]
I've had a deeper look into this issue using the report Ossama provided above. There are a few WCAG AA contrast failures that stem from the default Material theme colour palette.

**Expected behaviour:**

The [R Dev Container Docs](https://contributor.r-project.org/r-dev-env/) should meet WCAG AA contrast requirements so that the guidance is accessible to all contributors.

**Actual behaviour:**

The current palette (primary: `indigo`, accent: `blue`) introduces several contrast issues:

- Text colour `#7f7f7f` on background colour `#171717` giving contrast ratio 4.47:1 < 4.5:1 requirement. Affects the md-copyright element in the footer in both light and dark mode.
- Text colour `#717171` on background colour `#f5f5f5` giving contrast ratio 4.47:1 < 4.5:1 requirement. Affects lighter text in code blocks when in light mode e.g. `$TOP_SRCDIR` in the step 2 code block on [Building R tutorial page](https://contributor.r-project.org/r-dev-env/tutorials/building_r/).
- Link text colour `#4051b5` compared to surrounding text colour `#212121` giving contrast ratio 2.34:1 < 3:1 requirement. Affects links provided in the main body text in light mode.

**Suggested solutions:**

Either explore different colour combinations for the primary and accent colours to find one that meets WCAG AA in both light and dark mode (ensuring not to violate any other rules met by the current combination) or add a style sheet file to enable colour customisation of specific elements as detailed in [mkdocs-material/docs/customization.md](https://github.com/squidfunk/mkdocs-material/blob/master/docs/customization.md).

**Comment by rivaquiroga** [comment]
Thanks @soph-scott, this is super useful.
I'm  currently exploring migrating the guide to Quarto (https://github.com/r-devel/r-dev-env/issues/312), as that will make everything easier to maintain, since we can propagate the color contrast fix we decide on for the R Dev Guide and the R Contributors website issues to the Containers Docs.

**Comment by soph-scott** [comment]
Thanks @rivaquiroga that sounds like a good plan to me! I had a go at adjusting the colours to meet the WCAG standards that were being violated by the current palette choices. I've posted the code below on the off chance it is useful but it is quite hacky so not a very elegant solution! Moving everything into Quarto with a WCAG‑compliant colour scheme across all guidance sounds like a much better approach - with the added benefit of giving all the docs a consistent, professional theme. 😊

**Added to existing file mkdocs.yml:**
```
extra_css:
  - stylesheets/extra.css
```

**Added to new file docs\stylesheets\extra.css (comments included for context):**
```
/* To underline links in main body text but not code blocks/code line numbers */
.md-content__inner.md-typeset a:not(.md-typeset pre a) {
  text-decoration: underline;
}

/* To use blue for link colour and indigo for the accent colour in dark mode (inverse of light mode) */
[data-md-color-scheme="slate"] .md-typeset a {
  color: #6EA8FF;
}

[data-md-color-scheme="slate"] .md-typeset a:hover,
[data-md-color-scheme="slate"] .md-typeset a:focus {
  color: var(--md-accent-fg-color);
}

/* To lighten copyright text in footer */
.md-copyright {
  color: #A6A6A6;
}

/* To lighten links in footer to maintain a contrast with copyright text similar to before */
html .md-footer-meta.md-typeset a {
  color: #E6E6E6;
  text-decoration: none;
}

/* To improve contrast for troublesome code elements in light mode */
[data-md-color-scheme="default"] .md-typeset .nv,
[data-md-color-scheme="default"] .md-typeset pre code .go,
[data-md-color-scheme="default"] .md-typeset pre code a[href^="#__codelineno"],
[data-md-color-scheme="default"] .md-typeset .linenodiv a[href^="#__codelineno"] {
  color: #707070;
}

/* To improve contrast for troublesome code elements in dark mode */
[data-md-color-scheme="slate"] .md-typeset .nb {
  color: #A797E7;
}

[data-md-color-scheme="slate"] .md-typeset pre code .m {
  color: #ED8578;
}
```
